JuliaStream.jl/src/DistributedStream.jl

```diff
@@ -0,0 +1,85 @@
+using Distributed
+
+@everywhere using Pkg
+@everywhere Pkg.activate("."; io=devnull) # don't spam `Activating environment at...`
+@everywhere include("StreamData.jl")
+@everywhere include("Stream.jl")
+@everywhere using SharedArrays
+@everywhere const SharedArrayData = StreamData{T,SharedArray{T}} where {T}
+
+function devices()
+  return ["CPU (localhost)"]
+end
+
+function make_stream(
+  arraysize::Int,
+  scalar::T,
+  device::Int,
+  silent::Bool,
+)::SharedArrayData{T} where {T}
+  if device != 1
+    error("Only CPU device is supported")
+  end
+
+  if !silent
+    println("Using max $(nworkers()) process(es) + 1 master")
+  end
+  return SharedArrayData{T}(
+    SharedArray{T}(arraysize),
+    SharedArray{T}(arraysize),
+    SharedArray{T}(arraysize),
+    scalar,
+    arraysize,
+  )
+end
+
+function init_arrays!(data::SharedArrayData{T}, init::Tuple{T,T,T}) where {T}
+
+  @sync @distributed for i = 1:data.size
+    @inbounds data.a[i] = init[1]
+    @inbounds data.b[i] = init[2]
+    @inbounds data.c[i] = init[3]
+  end
+end
+
+function copy!(data::SharedArrayData{T}) where {T}
```

Comment:
Note that here and in the functions below where `T` isn't really used to constrain the signature nor in the body of the function, the parameter is redundant and
```suggestion
function copy!(data::SharedArrayData)
```
would work just fine